sw/airborne/firmwares/rotorcraft/guidance/guidance_indi_hybrid.c

```diff
@@ -651,6 +708,98 @@ struct FloatVect3 nav_get_speed_sp_from_line(struct FloatVect2 line_v_enu, struc
 
   return speed_sp_return;
 }
+/**
+ * @brief Go to a waypoint following a controlled diagonal descend towards a rope
+ *
+ * @param target the target waypoint
+ *
+ * @param rope_heading heading in radians of the landing rope calculated from the north
+ * 
+ * @return desired speed FloatVect3
+ */
+struct FloatVect3 nav_get_speed_sp_from_diagonal(struct EnuCoor_i target, float pos_gain, float rope_heading) {
+  // The speed sp that will be returned
+  struct FloatVect3 speed_sp_return = {0, 0, 0};
+  struct NedCoor_f ned_target_wp;
+
+  // Target in NED instead of ENU and altitude relative controller
+  VECT3_ASSIGN(ned_target_wp, POS_FLOAT_OF_BFP(target.y), POS_FLOAT_OF_BFP(target.x), -POS_FLOAT_OF_BFP(target.z));
+  //VECT3_ASSIGN(ned_target_wp, POS_FLOAT_OF_BFP(target.y), POS_FLOAT_OF_BFP(target.x), POS_FLOAT_OF_BFP(-nav_flight_altitude));
+  
+  // Calculate position error
+  struct FloatVect3 pos_error;
+  struct NedCoor_f *pos = stateGetPositionNed_f();
+  VECT3_DIFF(pos_error, ned_target_wp, *pos);
+
+  // Calculate normal vecotor to rope
+  // Rope Heading in degrees
+  struct FloatVect3 n_I;
+  float n_I_N = cosf(rope_heading);
+  float n_I_E = sinf(rope_heading);
+  float n_I_D = 0;
+  VECT3_ASSIGN(n_I,n_I_N,n_I_E,n_I_D);
+
+  // Rotate vector to have a glideslope angle
+  struct FloatVect3 l_I;
+  float c_omega = cosf(RadOfDeg(landing_slope));
```

Comment:
If you use non-standard units like degrees, please add _deg to the name

Reply:
yes thank you for the tip!